pallets/nfts/src/tests.rs

```diff
@@ -1992,6 +2291,118 @@ fn approvals_limit_works() {
 	});
 }
 
+#[test]
+fn approve_collection_transfer_works() {
+	new_test_ext().execute_with(|| {
+		let (collection, locked_collection) = (0, 1);
+		let item = 42;
+		let collection_owner = account(1);
+		let item_owner = account(2);
+		let delegate = account(3);
+
+		Balances::make_free_balance_be(&item_owner, 100);
+		Balances::make_free_balance_be(&delegate, 100);
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			collection_owner.clone(),
+			default_collection_config()
+		));
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			collection_owner.clone(),
+			default_collection_config()
+		));
+
+		// approve collection without items, throws error `Error::NoItemOwned`.
+		assert_noop!(
+			Nfts::approve_collection_transfer(
+				RuntimeOrigin::signed(item_owner.clone()),
+				collection,
+				delegate.clone(),
+				None
+			),
+			Error::<Test>::NoItemOwned
+		);
+
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(collection_owner.clone()),
+			collection,
+			item,
+			item_owner.clone(),
+			default_item_config()
+		));
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(account(1)),
+			locked_collection,
+			item,
+			item_owner.clone(),
+			default_item_config()
+		));
+
+		// throws error `Error::ItemsNonTransferable`.
+		assert_ok!(Nfts::lock_collection(
+			RuntimeOrigin::signed(account(1)),
```

Comment:
```suggestion
			RuntimeOrigin::signed(collection_owner),
```

Reply:
Please check this test as I see `account`s being used who have a variable for them...